presto-main/src/main/java/com/facebook/presto/sql/planner/planPrinter/WindowOperatorStats.java

```diff
@@ -0,0 +1,131 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.planner.planPrinter;
+
+import com.facebook.presto.operator.WindowOperator;
+import com.google.common.collect.ImmutableList;
+import io.airlift.stats.Distribution;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+
+import static com.google.common.collect.Iterables.concat;
+import static java.util.Objects.requireNonNull;
+
+class WindowOperatorStats
+{
+    private final List<Long> indexSize;
+    private final List<Long> partitionRowsPerIndex;
+    private final List<Long> partitionRowsPerDriver;
+    private final int activeDrivers;
+    private final int totalDrivers;
+
+    public static WindowOperatorStats create(WindowOperator.WindowInfo info)
+    {
+        ImmutableList.Builder<Long> indexSize = ImmutableList.builder();
+        ImmutableList.Builder<Long> partitionRowsPerIndex = ImmutableList.builder();
+        info.getWindowInfos().stream()
+                .map(WindowOperator.DriverWindowInfo::getIndexInfo)
+                .flatMap(Collection::stream)
+                .forEach(indexInfo -> {
+                    indexSize.add(indexInfo.getSizeInBytes());
+                    partitionRowsPerIndex.add(indexInfo.getTotalRowCountInPartitions());
+                });
+
+        ImmutableList.Builder<Long> partitionRowsPerDriver = ImmutableList.builder();
+        int activeDrivers = 0;
+        int totalDrivers = 0;
+
+        for (WindowOperator.DriverWindowInfo driverWindowInfo : info.getWindowInfos()) {
+            long partitionRowsCount = driverWindowInfo.getIndexInfo().stream()
+                    .mapToLong(WindowOperator.IndexInfo::getTotalRowCountInPartitions)
+                    .sum();
+            partitionRowsPerDriver.add(partitionRowsCount);
+            totalDrivers++;
+            if (partitionRowsCount > 0) {
+                activeDrivers++;
+            }
+        }
+
+        return new WindowOperatorStats(indexSize.build(), partitionRowsPerIndex.build(), partitionRowsPerDriver.build(), activeDrivers, totalDrivers);
+    }
+
+    private WindowOperatorStats(List<Long> indexSize, List<Long> partitionRowsPerIndex, List<Long> partitionRowsPerDriver, int activeDrivers, int totalDrivers)
+    {
+        this.indexSize = requireNonNull(indexSize, "indexSize is null");
+        this.partitionRowsPerIndex = requireNonNull(partitionRowsPerIndex, "partitionRowsPerIndex is null");
+        this.partitionRowsPerDriver = requireNonNull(partitionRowsPerDriver, "partitionRowsPerDriver is null");
+        this.activeDrivers = activeDrivers;
+        this.totalDrivers = totalDrivers;
+    }
+
+    public static Optional<WindowOperatorStats> merge(Optional<WindowOperatorStats> first, Optional<WindowOperatorStats> second)
```

Comment:
Discussed with @martint, he is OK with moving Mergeable to the util package and using that instead.